compiler/modules/front/codeGen.bal

```diff
@@ -1118,8 +1125,12 @@ function codeGenExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Ex
         var { opPos: pos, bitwiseOp: op, left, right } => {
             var { result: l, block: block1} = check codeGenExprForInt(cx, bb, env, left);
             var { result: r, block: nextBlock } = check codeGenExprForInt(cx, block1, env, right);
+            bir:Operand lhs;
+            bir:Operand rhs;
+            bir:BasicBlock? nilBlock;
+            { lhs, rhs, nextBlock, nilBlock } = codeGenBinaryNilLift(cx, l, r, nextBlock, pos);
```

Comment:
It's not really right to use codeGenExprForInt if the result might be `()` (you will get a misleading error message).  The codeGenNilLift should be taking a list of expressions not operands. Then you should check the type of the operand is int fter doing the nil lifting.

Reply:
fixed